jumpscale/sals/nginx/nginx.py

```diff
@@ -113,27 +114,46 @@ def get_config(self):
         return render_config_template("website", base_dir=j.core.dirs.BASEDIR, website=self)
 
     def generate_certificates(self):
-        if self.ssl:
-            j.sals.process.execute(
-                f"certbot --nginx -d {self.domain} --non-interactive --agree-tos -m {self.letsencryptemail} --nginx-server-root {self.parent.cfg_dir}"
-            )
+        res = j.sals.process.execute(
+            f"certbot --nginx -d {self.domain} --non-interactive --agree-tos -m {self.letsencryptemail} --nginx-server-root {self.parent.cfg_dir}"
+        )
+        if res[0] != 0:
+            if self.selfsigned:
+                res = self.generate_self_signed_certificates()
+                if res and res[0] != 0:
+                    raise j.exceptions.JSException(f"Failed to generate self-signed certificate.{res}")
+                j.sals.fs.write_file(self.cfg_file, self.get_config())
+                return res
+            else:
+                raise j.exceptions.JSException(f"Failed to generate certificates by certbot.{res}")
+        return res
```

Comment:
return is not really needed as you throw exception on failure
you might want to consider returning a boolean indicating if it's selfsigned or generated by certbot (document in docstring if you decide to do so)